src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const finalState = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (true) {
```

Comment:
using a switch with a boolean condition is not a good practice. You can set action.type as a condition.